python/cuml/decomposition/tsvd.pyx

```diff
@@ -317,37 +321,42 @@ class TruncatedSVD:
 
+        cdef cumlHandle* handle_ = <cumlHandle*><size_t>self.handle.getHandle()
         if not _transform:
             if self.gdf_datatype.type == np.float32:
-                tsvdFit(<float*> input_ptr,
-                               <float*> components_ptr,
-                               <float*> singular_vals_ptr,
-                               params)
+                tsvdFit(handle_[0],
+                        <float*> input_ptr,
+                        <float*> components_ptr,
+                        <float*> singular_vals_ptr,
+                        params)
             else:
-                tsvdFit(<double*> input_ptr,
-                               <double*> components_ptr,
-                               <double*> singular_vals_ptr,
-                               params)
+                tsvdFit(handle_[0],
+                        <double*> input_ptr,
+                        <double*> components_ptr,
+                        <double*> singular_vals_ptr,
+                        params)
         else:
             if self.gdf_datatype.type == np.float32:
-                tsvdFitTransform(<float*> input_ptr,
-                                        <float*> trans_input_ptr,
-                                        <float*> components_ptr,
-                                        <float*> explained_var_ptr,
-                                        <float*> explained_var_ratio_ptr,
-                                        <float*> singular_vals_ptr,
-                                        params)
+                tsvdFitTransform(handle_[0],
+                                 <float*> input_ptr,
+                                 <float*> trans_input_ptr,
+                                 <float*> components_ptr,
+                                 <float*> explained_var_ptr,
+                                 <float*> explained_var_ratio_ptr,
+                                 <float*> singular_vals_ptr,
+                                 params)
             else:
-                tsvdFitTransform(<double*> input_ptr,
-                                        <double*> trans_input_ptr,
-                                        <double*> components_ptr,
-                                        <double*> explained_var_ptr,
-                                        <double*> explained_var_ratio_ptr,
-                                        <double*> singular_vals_ptr,
-                                        params)
+                tsvdFitTransform(handle_[0],
+                                 <double*> input_ptr,
+                                 <double*> trans_input_ptr,
+                                 <double*> components_ptr,
+                                 <double*> explained_var_ptr,
+                                 <double*> explained_var_ratio_ptr,
+                                 <double*> singular_vals_ptr,
+                                 params)
 
         components_gdf = cudf.DataFrame()
```

Comment:
Line 363 access `self.components_` does this require a sync for the potentially asynchronous tsvd call?

Reply:
I have put a sync call. Thanks